howtos/deployment.rst

```diff
@@ -0,0 +1,348 @@
+        return ":".join(["%s/%s" % (varname, d) for d in dirs])
+
+    path = _format_dirs(bin_dirs.values())
+    ld_library_path = _format_dirs(bin_dirs.values())
+    exe = varname + "/" + executable
+
+    content = """#!/usr/bin/env bash
+    set -ex
+    export PATH=$PATH:{path}
+    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:{ld_library_path}
+    pushd $(dirname {exe})
+    $(basename {exe})
+    popd
+    """.format(path=path,
+           ld_library_path=ld_library_path,
+           exe=exe)
+
+.. note::
+
+    The full example might be found on `GitHub <https://github.com/SSE4/conan-deploy-tool>`_.
+
+Approaches
+----------
+
+system package manager
+----------------------
+
+Usually the recommended option is to use system package manager, like making **.rpm** or ***.deb** package(s). Unfortunately, 
+it's not always an option, as there are few limitations:
+
+- scalability problem - need to create one package per supported distro (likely one for `Ubuntu <https://ubuntu.com/>`_, one for `Arch Linux <https://www.archlinux.org/>`_, etc.), and formats or guidelines for each distro might differ significantly
+- it might be needed to (re)compile application N times for each of supported distro, or at least use the lowest version (see concerns about ``glibc`` below)
+- sometimes older distributions may lack packages for actual software releases (e.g. boost 1.71 might be missing in `CentOS <https://www.centos.org/?>`_ 6 or 7)
+- not all systems use package managers, for instance, some embedded systems or `LFS <http://www.linuxfromscratch.org/>`_ distributions
+
+As a result, there are other ways to help with application deployment.
+
+archives (tarballs)
+-------------------
+
+The most straightforward thing to do is to make an archive with application and all of its dependencies and resources, such as zip file, or tarball (``.tar.gz``/``.tgz``, ``.tar.bz``/``.tbz``, ``.tar.xz``/``.txz`` or just ``.tar`` file).
+The obvious disadvantage is usability: the user has to extract an archive first, and then locate an executable (or startup script) within the 
+extracted directory.
+
+makeself.io
+-----------
+
+`makeself <https://makeself.io>`_ is a small command-line utility to generate self-extracting archives.
+**makeself** is pretty popular and used for instance by 
+`VirtualBox <https://www.virtualbox.org/wiki/Linux_Downloads>`_ and 
+`CMake <https://cmake.org/download/>`_.
+Such archives are usually used to create installers, and they often have ``.run``, ``.bin`` or ``.sh`` extensions.
+Makeself created archives are just small startup scripts concatenated with tarballs.
+When you run such self-extracting archive:
+
+- small script (shim) extracts the embedded archive into the temporary directory
+- script passes the execution to the entry point within the unpacked archive
+- application is being run
+- finally, the temporary directory removed
+
+Therefore, it transparently appears just like a normal application execution.
```

Comment:
At least link to a getting started guide or something if we cannot provide an example.

Reply:
added link to http://xmodulo.com/how-to-create-a-self-extracting-archive-or-installer-in-linux.html